overlord/devicestate/handlers.go

```diff
@@ -57,6 +57,23 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	var modelass []byte
+	if err := t.Get("new-model", &modelass); err != nil {
+		return err
+	}
+	model, err := asserts.Decode(modelass)
+	if err != nil {
+		return err
+	}
+
+	return assertstate.Add(st, model)
```

Comment:
this fails we same assertion is added again,  we need to ignore that error to make this idempotent (there are examples about that in other places, let me know if you need help to locate one), we need tests too about running this twice